### PR TITLE
Added gaussian 2d conditional GAN and gaussian 1d problems and networks

### PR DESCRIPTION
**Pull request by jamaltoutouh**
I wave added new Conditional-GANs and I have created a toy gaussian 2d problems to address them by using CGANs. Besides, I have reformulated the gaussian 1d problem to allow people to run lipi with several clients on a simple laptop.

### CODE REVIEW

File: src/data/gaussian_2d_data_loader.py [hunk 1]
```diff
@@ -2,55 +2,100 @@
...
     def save_images(self, images, shape, filename):
         self.dataset().save_images(images, filename)
 
+    @property
+    def points(self):
+        pass
+
 
-class GridToyDataLoader(DataLoader):
+class UnlabeledGridToyDataLoader(DataLoader):
```
Comment (hembergerik) [inline src/data/gaussian_2d_data_loader.py:84]: Seems very similar to `LabeledGridDataLoader` can it subclass it?

File: src/data/gaussian_2d_data_loader.py [hunk 2]
```diff
@@ -2,55 +2,100 @@
...
+
+    @property
+    def num_classes(self):
+        return self.dataset().num_classes
+
+    def save_images(self, images, shape, filename):
+        self.dataset().save_images(images, filename)
+
+
+class UnlabeledCircularToyDataLoader(DataLoader):
```
Comment (hembergerik) [inline src/data/gaussian_2d_data_loader.py:40]: Seems very similar to `LabeledCirculrDataLoader` can it subclass it?

File: src/data/gaussian_2d_data_loader.py [hunk 3]
```diff
@@ -63,28 +108,35 @@ def points(self):
...
-            "number_of_modes"
-        ] = number_of_modes  # If does not exists it creates the parameter
-        xs, ys = self.points(number_of_modes)
-        points_array = np.array((xs, ys), dtype=np.float).T
+        xs, ys, labels = self.points(number_of_modes)
+        points_with_label = np.array((xs, ys, labels), dtype=np.float).T
+        points_with_label = points_with_label[np.random.choice(points_with_label.shape[0], self.number_of_records), :]
+        points_array = [[point[0], point[1]] for point in points_with_label]
+
+        # self.labels = [self.create_label(int(point[2])) for point in points_with_label]
```
Comment (hembergerik) [inline src/data/gaussian_2d_data_loader.py:123]: Remove commented code if not used

**Review by hembergerik — APPROVED** [review]

File: src/data/gaussian_2d_data_loader.py (hunk 2 above)
Comment (jamaltoutouh) [reply src/data/gaussian_2d_data_loader.py]: Thanks. I have added subclasses for DataLoaders and DataSets.

File: src/data/gaussian_2d_data_loader.py (hunk 3 above)
Comment (jamaltoutouh) [reply src/data/gaussian_2d_data_loader.py]: I have removed it.

**Review by hembergerik — APPROVED** [review]

**Pull request merged**